flink-core/src/main/java/org/apache/flink/api/connector/source/SourceReaderContext.java

```diff
@@ -68,8 +79,24 @@ public interface SourceReaderContext {
      * Get the current parallelism of this Source.
      *
      * @return the parallelism of the Source.
+     * @deprecated This method is deprecated since Flink 1.19. All metadata about the task should be
+     *     provided uniformly by {@link #getTaskInfo()}.
+     * @see <a
+     *     href="https://cwiki.apache.org/confluence/display/FLINK/FLIP-382%3A+Unify+the+Provision+of+Diverse+Metadata+for+Context-like+APIs">
+     *     FLIP-382: Unify the Provision of Diverse Metadata for Context-like APIs </a>
      */
+    @Deprecated
     default int currentParallelism() {
         throw new UnsupportedOperationException();
     }
+
+    /**
+     * Get the meta information of current task.
+     *
+     * @return the task meta information.
+     */
+    @PublicEvolving
+    default TaskInfo getTaskInfo() {
+        return null;
```

Comment:
Same here for the default implementation and Nullable.

Reply:
I've changed the default methods including `getJobInfo()` and `getTaskInfo()` to be abstract.